atc/creds/vault/vault_test.go

```diff
@@ -31,6 +36,24 @@ func (msr *MockSecretReader) Read(lookupPath string) (*vaultapi.Secret, error) {
 	return nil, nil
 }
 
+func createMockV2Secret(value string) *vaultapi.Secret {
+	escapedInput, err := json.Marshal(&value)
+	Expect(err).To(BeNil())
+	secret := vaultapi.Secret{}
+	json.Unmarshal([]byte(fmt.Sprintf(`{"data":{"value":%s},"metadata":{"created_time":"2021-01-06T22:32:10.969537Z","deletion_time":"","destroyed":false,"version":3}}`, escapedInput)), &secret.Data)
```

Comment:
Seems like this could be a [`vaultapi.Secret`](https://pkg.go.dev/github.com/hashicorp/vault/api#Secret) literal:

```go
return &vaultapi.Secret{
  Data: map[string]interface{}{
    "data": map[string]interface{}{"value": value},
    "metadata": map[string]interface{}{
      "created_time": "2021-01-06T22:32:10.969537Z",
      "deletion_time": "",
      "destroyed": false,
      "version": 3,
    },
  },
}
```